DataFormats/HcalRecHit/interface/HBHEChannelInfo.h

```diff
@@ -18,18 +18,26 @@ class HBHEChannelInfo
 
     static const unsigned MAXSAMPLES = 10;
 
-    inline HBHEChannelInfo()
-      : rawCharge_{0.}, pedestal_{0.}, pedestalWidth_{0.},
-        gain_{0.}, gainWidth_{0.}, riseTime_{0.f}, adc_{0},
-        dFcPerADC_{0.f}, hasTimeInfo_(false), hasEffectivePedestals_(false) {clear();}
-
-    inline explicit HBHEChannelInfo(const bool hasTimeFromTDC, const bool hasEffectivePed)
-      : rawCharge_{0.}, pedestal_{0.}, pedestalWidth_{0.},
-        gain_{0.}, gainWidth_{0.}, riseTime_{0.f}, adc_{0},
-        dFcPerADC_{0.f}, hasTimeInfo_(hasTimeFromTDC), 
-        hasEffectivePedestals_(hasEffectivePed) {clear();}
-
-    inline void clear()
+    constexpr HBHEChannelInfo()
+      : id_(), rawCharge_{0.}, pedestal_{0.}, pedestalWidth_{0.},
+        gain_{0.}, gainWidth_{0.}, darkCurrent_{0}, fcByPE_{0}, lambda_{0},
+        riseTime_{0.f}, adc_{0},
+        dFcPerADC_{0.f}, recoShape_{0}, nSamples_{0}, 
+        soi_{0}, capid_{0}, hasTimeInfo_{false}, hasEffectivePedestals_{false},
+        dropped_{true}, hasLinkError_{false}, hasCapidError_{false}
+    {clear();}
```

Comment:
here and below the ```clear()``` is redundant and can be dropped

Reply:
can be, i'm trying to leave things as is as much as possible. I will remove it